### DISCUSSION TITLE
Reasons to not use "client: false" in move functions?

### DISCUSSION THREAD

**Discussion by TinyTakinTeller** [boardgameio/boardgame.io#1202] [Q&A] [answered]
Why would you ever not use `client: false` in your move functions? (Why is this not the default?)

https://boardgame.io/documentation/#/secret-state?id=disabling-moves-that-manipulate-secret-state-on-the-client

_I'm trying to wrap my head around on how exactly G is handled on client / server._

**Comment by delucis** [comment, marked as answer]
Some games do not have secret state. For example, think of chess: all information about the game state is publicly visible to both players. `client: false` means the player has to wait for a server response to see the result of their action, so when it is OK to do so, calculating on the client is a snappier user experience because they get immediate feedback about their move.

In games with secret state (decks of face down cards, player state that is not visible to other players, etc.) then the move can’t run on the client, so that’s why `client: false` is needed.

**Reply by TinyTakinTeller** [reply]
Oh, optimistic updates https://boardgame.io/documentation/#/multiplayer?id=clients-and-masters

Thanks!
